include/GafferImage/ImageReader.h

```diff
@@ -110,10 +110,18 @@ class ImageReader : public ImageNode
 		Gaffer::StringPlug *colorSpacePlug();
 		const Gaffer::StringPlug *colorSpacePlug() const;
 
+		Gaffer::StringPlug *colorSpaceMetadataNamePlug();
```

Comment:
Yes, I think you would use the OpenImageIOReader.out.metadata plug. You could just call `getValue()` on it and then add it to your CompoundData I think :

```
compoundData->writable()["metadata"] = imagePlug->metadataPlug()->getValue();
```